packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/properties-section/property-table/cells/value-cell/value-cell-editor/number-or-string-editor.tsx

```diff
@@ -0,0 +1,28 @@
+import { TextField } from "@hashintel/hash-design-system";
+import { cloneDeep } from "lodash";
+import { ValueCellEditorProps } from "../types";
+
+export const NumberOrStringEditor: ValueCellEditorProps = ({
+  value: cell,
+  onChange,
+}) => {
+  const { value } = cell.data.property;
+  const isNumber = typeof value === "number";
+
+  return (
+    <TextField
+      sx={{ my: "1px" }}
+      autoFocus
+      value={value}
+      type={isNumber ? "number" : "text"}
+      inputMode={isNumber ? "numeric" : "text"}
+      onChange={({ target }) => {
+        const newCell = cloneDeep(cell);
+        const newValue = isNumber ? Number(target.value) : target.value;
```

Comment:
This makes it impossible to use the keyboard to delete the last number in the cell.